compiler/rustc_infer/src/infer/higher_ranked/mod.rs

```diff
@@ -24,9 +24,9 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
         // as-is, we need to do some extra work here in order to make sure
         // that function subtyping works correctly with respect to regions
         //
-        // Note: this is a subtle algorithm.  For a full explanation,
-        // please see the large comment at the end of the file in the (inlined) module
-        // `doc`.
+        // Note: this is a subtle algorithm.  For a full explanation, please see
+        // the rustc development guide:
```

Comment:
It's not a big deal, but people almost always refer to it as the 'rustc-dev-guide' or 'rustc dev guide'. So it's possible that it may confuse people to use a longer name. But, like I said it's not a big deal :)

Reply:
Wow, you're fast :)